tests/chocolatey-tests/features/EnvironmentVariables.Tests.ps1

```diff
@@ -0,0 +1,44 @@
+﻿Describe "Ensuring Chocolatey Environment variables are correct" -ForEach @(
+    "config"
+    "cli"
+) -Tag EnvironmentVariables, Chocolatey {
+    BeforeDiscovery {
+        $TestedVariables = @(
+            @{ Name = 'TEMP' ; Value = "C:\Temp\$PID" }
+            @{ Name = 'TMP' ; Value = "C:\Temp\$PID" }
+            @{ Name = 'ChocolateyPackageFolder' ; Value = '{0}\chocolatey\lib\test-environment' }
+            @{ Name = 'ChocolateyPackageName' ; Value = 'test-environment' }
+            @{ Name = 'ChocolateyPackageTitle' ; Value = 'test-environment (Install)' }
+            @{ Name = 'ChocolateyPackageVersion' ; Value = '1.0.0' }
+        )
+    }
+
+    BeforeAll {
+        Initialize-ChocolateyTestInstall
+        New-ChocolateyInstallSnapshot
+        $cacheGuid = New-Guid
+        $cacheDir = "$(Get-TempDirectory)/$cacheGuid"
+        switch ($_) {
+            'config' {
+                Invoke-Choco config set --name=cachelocation --value $cacheDir
+            }
+            'cli' {
+                $cacheArg = "--cachedir='$cacheDir'"
+            }
+        }
+        $Output = Invoke-Choco install test-environment --version 1.0.0 $cacheArg

```

Comment:
@corbob just to note in case you have a need for it.

I have pushed another version of this package to the test packages repository.
It uses a non-normalized version of `0.9` (needed it for my own tests), and it also outputs environment variables during before modify, and uninstalls as well.